starter-code/src/viking.js

```diff
@@ -1,11 +1,104 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+Soldier.prototype.attack = function(){
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage){
+
+ this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+Viking.prototype.receiveDamage = function(damage){
+
+  this.health -= damage;
+
+  if(this.health <= 0){
+      return this.name + " has died in act of combat";
+  } else { return this.name + " has received "+ damage +" points of damage"}
+}
+
+Viking.prototype.battleCry = function(){
+  return 'Odin Owns You All!'
+}
+
+function Viking(name, health, strength){
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
 
 // Saxon
-function Saxon() {}
 
-// War
-function War() {}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+Saxon.prototype.receiveDamage = function(damage){
+
+  this.health -= damage;
+
+  if(this.health <= 0){
+      return "A Saxon has died in combat";
+  } else { return  "A Saxon has received "+ damage +" points of damage"}
+}
+
+function Saxon(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+
+// BONUS: WAR!!
+function War() {
+  this.vikingArmy= [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking= function (viking){
+  this.vikingArmy.push(viking)
+};
+
+War.prototype.addSaxon = function(saxon){
+  this.saxonArmy.push(saxon)
+}
+
+
+War.prototype.vikingAttack= function(){
+
+  var randomSaxon = this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)];//random index of saxonArmy
+  var randomViking = this.vikingArmy[Math.floor(Math.random()*this.vikingArmy.length)];//random index of vikingArmy
+
+  var vikingAttackReturn = randomSaxon.receiveDamage(randomViking.strength);
+
+  this.saxonArmy = this.saxonArmy.filter(function(saxon){
+          return saxon.health > 0;
```

Comment:
Utilizar `filter` aquí no es correcto, ya que estarás iterando sobre todo el array de los vikingos o de los sajones independientemente alguno haya muerto o no. Porque no tienen por qué morir tras un ataque. Es más correcto si evalúas la vida del vikingo al que acabas de atacar, y si la vida es 0, entonces quitáis a ese vikingo del array.